app/models/repository.rb

```diff
@@ -10,6 +12,9 @@ class Repository < ActiveRecord::Base
   belongs_to :parent,   class_name: "Repository", foreign_key: "make_id"
   paginates_per 12
 
+  # default_scope { where(:share_type => "public") }
+  scope :public_repos, -> { where(share_type:"public") }
```

Comment:
If the user is looking at his/her own profile, they need to see the private ones too. But if it's someone else looking at their profile, they should _not_ see the private repos. the problem is that with the default_scope, @user.repositories only returns the public repos.